### PR TITLE
fix(tests): Run E2E tests on Pull Requests

### PR DESCRIPTION
**Pull request by jarifibrahim**
This PR adds planner E2E tests.
Once this PR is merged, each PR to planner repo will be tested against live backend (currently we use mock backend). 

**How to Run tests locally**
1. Login to prod-preview.openshift.io
2. Open Console (F12). Click on `Application` tab, go to `Local Storage` => `https://prod-preview.openshift.io`. Copy the value of `auth_token` key. Let's call this value `TOKEN`
3. Run `BASE_URL=prod-preview.openshift.io REFRESH_TOKEN=TOKEN USER_NAME=your_user_name DEBUG=true ./run_e2e_tests.sh`

NOTE - Please set the `BASE_URL` to your PR's deployed snapshot URL. The tests would fail if you run against `prod-preview.openshift.io` since all the changes in planner master are not on prod-preview.

### CODE REVIEW

**Comment by rgarg1** [comment]
Intent? @jarifibrahim 

Rather, can we add a few selective planner tests (from smoke suite) to Len's EE tests?

**Comment by jarifibrahim** [comment]
@rgarg1, we already spoke about this. 
The intent is to discard the mock based test and run the E2E tests that use the live backend against each PR.

**Comment by rgarg1** [comment]
@jarifibrahim Disregard my question. The title made be to believe that we are trying to run Len's tests on our PRs :)

**Comment by jarifibrahim** [comment]
[test]

**Comment by jarifibrahim** [comment]
[test]

**Comment by jarifibrahim** [comment]
[test]

**Comment by jarifibrahim** [comment]
[test]

File: cico_run_tests.sh [hunk 1]
```diff
@@ -14,7 +14,7 @@ set -e
 # that might interest this worker.
 if [ -e "jenkins-env" ]; then
   cat jenkins-env \
-    | grep -E "(JENKINS_URL|DEVSHIFT_USERNAME|DEVSHIFT_PASSWORD|GIT_BRANCH|GIT_COMMIT|BUILD_NUMBER|ghprbSourceBranch|ghprbActualCommit|BUILD_URL|ghprbPullId)=" \
+    | grep -E "^(JENKINS_URL|REFRESH_TOKEN|DEVSHIFT_USERNAME|DEVSHIFT_PASSWORD|GIT_BRANCH|GIT_COMMIT|BUILD_NUMBER|ghprbSourceBranch|ghprbActualCommit|BUILD_URL|ghprbPullId)=" \
```
Comment (pranavgore09) [inline cico_run_tests.sh]: why do we need `^` at start there? build fails with syntax error. `/tmp/jenkins-env: line 11: syntax error near unexpected token `('`

Comment (pranavgore09) [reply cico_run_tests.sh]: and please append new keys, easier to review :)

Comment (jarifibrahim) [reply cico_run_tests.sh]: >why do we need ^ at start there? build fails with syntax error. /tmp/jenkins-env: line 11: syntax error near unexpected token('`

Please check - https://chat.openshift.io/developers/pl/8cq4htg5y7fazcfogs46cpzf7a
and https://gitlab.cee.redhat.com/dtsd/housekeeping/issues/1679

**Comment by jarifibrahim** [comment]
[test]

**Comment by jarifibrahim** [comment]
[test]

**Review by Raunak1203 — APPROVED** [review]

**Comment by jarifibrahim** [comment]
[test]

**Pull request merged**